examples/cis3-nft-sponsored-txs/src/lib.rs

```diff
@@ -770,65 +770,18 @@ fn contract_transfer<S: HasStateApi>(
     Ok(())
 }
 
-/// Helper function to calculate the message hash used in the permit function.
-///
-/// It rejects if:
-/// - It fails to parse the parameter.
-/// - The message was intended for a different contract.
-/// - The message was intended for a different `entry_point`.
-/// - The message is expired.
-
+/// Helper function that can be invoked to serializ the `PermitMessage`.
 #[receive(
     contract = "cis3_nft",
-    name = "calculateMessageHash",
+    name = "serializationHelper",
     parameter = "PermitMessage",
-    return_value = "[u8;32]",
-    crypto_primitives,
     mutable
 )]
-fn contract_calculate_message_hash<S: HasStateApi>(
-    ctx: &impl HasReceiveContext,
+fn contract_serialization_helper<S: HasStateApi>(
```

Comment:
I don't understand the point of this receive function.
If it is needed, consider adding more to the description

Reply:
This function is only meant as a method to serialize the `PermitMessage` at the front-end before signing it in the wallet. It is technically a `view` function that should return nothing except for having an input parameter (the `PermitMessage`).

https://github.com/Concordium/concordium-browser-wallet/blob/df9e3cb8672d0cc7dabf6f64836b699ece53d15b/examples/sponsoredTXs/src/sponsoredTx.tsx#L176

Reply:
I removed the `mutative` and updated the explanation:
```
Helper function that can be invoked at the front-end to serialize the `PermitMessage` before signing it in the wallet.
```

Reply:
I will just hijack this conversation (I also find this functions rather puzzling and I don't understand why it is needed).

There is a function in the JS helpers which takes a schema and JSON and serializes it into a buffer.

Reply:
We need the `PermitMessage` in the schema and therefore it needs to be an input parameter of some function in the smart contract (currently there is no function using the plain `PermitMessage` as input). We could create the schema (so we have it for the front-end) and then remove this function from the smart contract again. I just think that it might be easier to understand for outside people if they see where we got this `schema` (that serializes the `PermitMessage`) comes from and hence leaving this dummy `serializingHelper` in the smart contract code. But I can also remove it.

Reply:
Right, I missed that. I read both as having PermitParams, sorry about that.